python/cudf/cudf/core/column/categorical.py

```diff
@@ -1456,11 +1457,7 @@ def _set_categories(
         # Ensure new_categories is unique first
         if not (is_unique or new_cats.is_unique):
             # drop_duplicates() instead of unique() to preserve order
-            new_cats = (
-                cudf.Series(new_cats)
-                .drop_duplicates(ignore_index=True)
-                ._column
-            )
+            new_cats = dedup_preserve_order(cudf.Series(new_cats))._column
```

Comment:
Can we have `dedup_preserve_order` operate on a `Column` rather than a `Series`? As  much as possible, we want to avoid  this pattern of constructing a `Series` out of a column, invoking a  Series operation on it, and then getting the column back out of the result.

Having it operate on a column would have the added benefit that we can then  make it a method of  `CategoricalColumn`.

Reply:
@shwina those are all good points, I'll update as such.

Reply:
I added this to `ColumnBase` in the end as I ended up not being able to see any reason all columns shouldn't share the capability. Let me know if this should be changed.